Brocade-Napalm Driver implementation

shivharis opened the pull request:
Please review and let me know your comments.

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+
+        lines = output.splitlines()
+
+        while True:
```
I have a problem with `break` statements:

Can define something like:

```
def _search_cpu_load(lines):

    cpu_regex = r'^.*One minute: (\d+\.\d+); Five.*$'
    for line in lines:
        if 'Realtime Statistics' in line:
            match = re.search(cpu_regex, line)
            if match:
                return match.group(1)
```

And use it:

```
cpu_usage = _search_cpu_load(lines)
if cpu_usage:
    environment['cpu'][0]['%load'] = cpu_usage
```

dbarrosop replied:
yeah, logic is a bit confusing, follow mircea's advise here : )

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+from shutil import copyfile
+
+
+# TBD(shh) Put this in config file (oslo_config)
```
Definitely remove this - don't need to have any hardcoded info, not even in a config file.

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                                         username=self.username,
+                                         password=self.password,
+                                         timeout=self.timeout,
+                                         verbose=True)
```
Why verbose?

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+        """
+        cli_output = dict()
+
+        if type(commands) is not list:
```
Better:

`if not isinstance(commands, list):`

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+        output = output[2:-1]
+
+        for line in output:
+            if len(line) == 0:
```
Instead of

``` python
+            if len(line) == 0:
 +                return {}
 +            if len(line.split()) == 6:
 +                address, mac, interface, macresolved, age, typ = line.split()
```

Can do:

``` python
try:
    address, mac, interface, macresolved, age, typ  = line.split()
except ValueError:
    continue  # will jump to next entry
```

Which avoids doing the same operation twice.

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                    'interface': interface,
+                    'mac': mac,
+                    'ip': address,
+                    'type': typ,
```
Field type should not exist: https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/test/models.py#L161

``` python
arp_table = {
    'interface': unicode,
    'mac': unicode,
    'ip': unicode,
    'age': float
}
```

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                        )
+                entry = {
+                    'interface': interface,
+                    'mac': mac,
```
Do instead:

`napalm_base.helpers.mac(mac)` to convert to NAPALM-specific MAC format.

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                return {}
+            if len(line.split()) == 6:
+                address, mac, interface, macresolved, age, typ = line.split()
+                try:
```
Not worth raising error for that - it may happen to not have the age of some entries.

Do instead: `napalm_base.helpers.convert(age, 0.0)` which will default to 0.0 when age is not valid.

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                entry = {
+                    'interface': interface,
+                    'mac': mac,
+                    'ip': address,
```
Same with ip:

```
napalm_base.helpers.ip(address)
```

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+
+    def rollback_config(self):
+
+        print "Reloading previous checkpoint ..."
```
Remove

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                    line.split()
+                entry = {
+                    'vlanid': vlanid,
+                    'mac+address': mac_address,
```
`mac_address` instead of `mac+address`

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                vlanid, tt, mac_address, type, state, port_type, port = \
+                    line.split()
+                entry = {
+                    'vlanid': vlanid,
```
Just `vlan` is the name of the field

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                    'mac+address': mac_address,
+                    'type': type,
+                    'state': state,
+                    'port_type': port_type,
```
Field `port_type` should not exist: https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/test/models.py#L194

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                    'vlanid': vlanid,
+                    'mac+address': mac_address,
+                    'type': type,
+                    'state': state,
```
`state` does not exist either: https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/test/models.py#L194

mirceaulinic commented on napalm_brocade/brocade.py~:
```diff
@@ -0,0 +1,428 @@
+#
```
Remove this file, please

mirceaulinic commented on napalm_brocade/utils/interactive/interactive.py:
```diff
@@ -0,0 +1,183 @@
+# Sample script to demonstrate loading a config for a device.
```
Remove this file please, this should be covered in the tests.

mirceaulinic commented on napalm_brocade/__init__.py:
```diff
@@ -0,0 +1,16 @@
+# the License.
+
+"""napalm_skeleton package."""
+from skeleton import SkeletonDriver
```
`from brocade import BrocadeDriver`

mirceaulinic commented on MANIFEST.in:
```diff
@@ -0,0 +1,3 @@
+include requirements.txt
+include napalm_brcd/utils/*.py
```
Did not see any util script, what do you need to include?

mirceaulinic commented on napalm_brocade/__init__.py:
```diff
@@ -0,0 +1,16 @@
+# License for the specific language governing permissions and limitations under
+# the License.
+
+"""napalm_skeleton package."""
```
`"""napalm_brocade package."""`

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+                                         password=self.password,
+                                         timeout=self.timeout,
+                                         verbose=True)
+        except Exception:
```
Catch something less generic than `Exception` - `NetMikoAuthenticationException` and `NetMikoTimeoutException` is probably what you are looking for.

mirceaulinic commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+        cmd = "show system"
+        fact_table = {}
+        output = self.device.send_command(cmd)
+        output = output.split('\n')
```
Can use:

``` python
output = output.splitlines()
```

dbarrosop commented on README.md:
```diff
@@ -1 +1,17 @@
+Please see instuction at:
+https://napalm.readthedocs.io/en/latest/
+
+##Caveats
```
This should be send to the `napalm` repo with the rest of the drivers' caveats.

https://github.com/napalm-automation/napalm/tree/develop/docs/support

dbarrosop commented on README.md:
```diff
@@ -1 +1,17 @@
+
+Brocade switches always perform a commit at the end of each CLI. Hence there is no explict "commit" command at the CLI level. The implementation here simulates a commit operation.
+
+When a "merge" is requested through the napalm (load_merge) the file is actually pushed to the switch, however it is not applied to the switch. It is stored on the switch as a _candidate.cfg file. When a commit is requestred this file is merged with the running config. The _candidate.cfg is deleted from the switch. A discard would have deleted this file without commiting the operations.
```
Great! : ) We will have to add some tests with VMs to ensure the workflow is consistent with the rest. Can you add a vagrant box to: https://github.com/napalm-automation/napalm-base/blob/develop/vagrant/Vagrantfile

dbarrosop commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+        # Initialize 'power', 'fan', and 'temperature' to default values
+        # (not implemented)
+
+        environment.setdefault('power', {})
```
If you are initializing `environment['power']`, please, use `environment['power'] = {}` or even use a `defaultdict(dict)`. `environment.setdefault` usually means the variable might or might not have been set beforehand, which I think it will never be the case in your code.

dbarrosop commented on napalm_brocade/brocade.py:
```diff
@@ -0,0 +1,428 @@
+            'output': -1.0,
+            'capacity': -1.0
+            }
+        environment.setdefault('fans', {})
```
same as above.

dbarrosop commented:
Can you also copy the following files from this repo?

https://github.com/napalm-automation/napalm-eos

.gitignore
setup.cfg
requirements-dev.txt

Once the minor changes requested have been addressed I think we can merge the PR and start doing some tests and requiring some minor changes before releasing.

shivharis commented:
Mircea,

Thanks, I will work things the way you suggested and send out a new pull.

(Can you please let me know if you are getting my responses to the "reply"
to this email)

Thanks,

Shiv

On Tue, Oct 11, 2016 at 1:16 AM, David Barroso notifications@github.com
wrote:

> Can you also copy the following files from this repo?
> 
> https://github.com/napalm-automation/napalm-eos
> 
> .gitignore
> setup.cfg
> requirements-dev.txt
> 
> Once the minor changes requested have been addressed I think we can merge
> the PR and start doing some tests and requiring some minor changes before
> releasing.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> https://github.com/napalm-automation/napalm-brocade/pull/1#issuecomment-252843266,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ADCK1_fzTtAfX5HzUWXQtjd5CF5SKbbxks5qy0XMgaJpZM4KTEs8
> .

## 

-Shiv

shivharis commented:
David,

Thanks, I will work things the way you suggested and send out a new pull.

Shiv

On Mon, Oct 10, 2016 at 4:30 PM, Mircea Ulinic notifications@github.com
wrote:

> _@mirceaulinic_ requested changes on this pull request.
> 
> I left some inline comments - to not spam even more please go through your
> code and make sure the following:
> - use napalm_base.helpers.mac and napalm_base.helpers.ip to ensure the
>   same output format across all drivers
> - when a line does not contain the right format, either throw error,
>   either handle it and jump to next entry (depending on the severity). Right
>   now you just return empty output. That's not OK.
> - go through the models file: https://github.com/napalm-
>   automation/napalm-base/blob/develop/napalm_base/test/models.py
>   https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/test/models.py
>   and return only those fields
> 
> ## When you fix these, we can move on and start writing some tests.
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        return output
> >   +
> > -    def get_environment(self):
> >   +
> > -        environment = {}
> > -        cpu_cmd = 'show proc cpu'
> >   +
> > -        output = self.device.send_command(cpu_cmd)
> > -        output = output.strip()
> > -        environment.setdefault('cpu', {})
> > -        environment['cpu'][0] = {}
> > -        environment['cpu'][0]['%load'] = 0.0
> >   +
> > -        lines = output.splitlines()
> >   +
> > -        while True:
> 
> I have a problem with break statements:
> 
> Can define something like:
> 
> def _search_cpu_load(lines):
> 
> ```
> cpu_regex = r'^.*One minute: (\d+\.\d+); Five.*$'
> for line in lines:
>     if 'Realtime Statistics' in line:
>         match = re.search(cpu_regex, line)
>         if match:
>             return match.group(1)
> ```
> 
> And use it:
> 
> cpu_usage = _search_cpu_load(lines)
> if cpu_usage:
>     environment['cpu'][0]['%load'] = cpu_usage
> 
> ---
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > +"""
> > +Brocade-Napalm Driver.
> > +
> > +This driver is meant for SLX and NOS based switches.
> > +"""
> > +from netmiko import ConnectHandler
> > +from napalm_base.base import NetworkDriver
> > +from napalm_base.exceptions import ConnectionException, MergeConfigException, \
> > -    ReplaceConfigException, SessionLockedException, CommandErrorException
> >   +
> >   +import os
> >   +import re
> >   +from shutil import copyfile
> >   +
> >   +
> >   +# TBD(shh) Put this in config file (oslo_config)
> 
> Definitely remove this - don't need to have any hardcoded info, not even
> 
> ## in a config file.
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        self.username = username
> > -        self.password = password
> > -        self.timeout = timeout
> > -        self.port = optional_args.get('port', 22)
> >   +
> >   +
> > -    def open(self):
> > -        """Open a connection to the device."""
> > -        try:
> > -            self.device = ConnectHandler(device_type='brocade_vdx',
> > -                                         ip=self.hostname,
> > -                                         port=self.port,
> > -                                         username=self.username,
> > -                                         password=self.password,
> > -                                         timeout=self.timeout,
> > -                                         verbose=True)
> 
> ## Why verbose?
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -            raise ConnectionException("Cannot connect to switch: %s:%s" \
> > -                                          % (self.hostname, self.port))
> >   +
> > -    def close(self):
> > -        """Close the connection to the device."""
> > -        self.device.disconnect()
> >   +
> > -    def cli(self, commands=None):
> > -        """
> > -        Execute a list of commands and return the output in a dictionary format using the
> > -        command as the key.
> >   +
> > -        """
> > -        cli_output = dict()
> >   +
> > -        if type(commands) is not list:
> 
> Better:
> 
> ## if not isinstance(commands, list):
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > +
> > -    def get_arp_table(self):
> > -        """
> > -        Get ARP table.
> > -        """
> > -        arp_table = []
> >   +
> > -        arp_cmd = 'show arp'
> > -        output = self.device.send_command(arp_cmd)
> > -        output = output.split('\n')
> >   +
> > -        # Skip the first two lines which is the header
> > -        output = output[2:-1]
> >   +
> > -        for line in output:
> > -            if len(line) == 0:
> 
> Instead of
> -            if len(line) == 0:
>   -                return {}
>   -            if len(line.split()) == 6:
>   -                address, mac, interface, macresolved, age, typ = line.split()
> 
> Can do:
> 
> try:
>     address, mac, interface, macresolved, age, typ  = line.split()
> except ValueError:
>     continue  # will jump to next entry
> 
> ## Which avoids doing the same operation twice.
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -                return {}
> > -            if len(line.split()) == 6:
> > -                address, mac, interface, macresolved, age, typ = line.split()
> > -                try:
> > -                    if age == '-':
> > -                        age = 0
> > -                    age = float(age)
> > -                except ValueError:
> > -                    print(
> > -                        "Unable to convert age value to float: {}".format(age)
> > -                        )
> > -                entry = {
> > -                    'interface': interface,
> > -                    'mac': mac,
> > -                    'ip': address,
> > -                    'type': typ,
> 
> Field type should not exist: https://github.com/napalm-
> automation/napalm-base/blob/develop/napalm_base/test/models.py#L161
> 
> arp_table = {
>     'interface': unicode,
>     'mac': unicode,
>     'ip': unicode,
>     'age': float
> }
> 
> ---
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        for line in output:
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 6:
> > -                address, mac, interface, macresolved, age, typ = line.split()
> > -                try:
> > -                    if age == '-':
> > -                        age = 0
> > -                    age = float(age)
> > -                except ValueError:
> > -                    print(
> > -                        "Unable to convert age value to float: {}".format(age)
> > -                        )
> > -                entry = {
> > -                    'interface': interface,
> > -                    'mac': mac,
> 
> Do instead:
> 
> ## napalm_base.helpers.mac(mac) to convert to NAPALM-specific MAC format.
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        """
> > -        arp_table = []
> >   +
> > -        arp_cmd = 'show arp'
> > -        output = self.device.send_command(arp_cmd)
> > -        output = output.split('\n')
> >   +
> > -        # Skip the first two lines which is the header
> > -        output = output[2:-1]
> >   +
> > -        for line in output:
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 6:
> > -                address, mac, interface, macresolved, age, typ = line.split()
> > -                try:
> 
> Not worth raising error for that - it may happen to not have the age of
> some entries.
> 
> Do instead: napalm_base.helpers.convert(age, 0.0) which will default to
> 
> ## 0.0 when age is not valid.
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 6:
> > -                address, mac, interface, macresolved, age, typ = line.split()
> > -                try:
> > -                    if age == '-':
> > -                        age = 0
> > -                    age = float(age)
> > -                except ValueError:
> > -                    print(
> > -                        "Unable to convert age value to float: {}".format(age)
> > -                        )
> > -                entry = {
> > -                    'interface': interface,
> > -                    'mac': mac,
> > -                    'ip': address,
> 
> Same with ip:
> 
> napalm_base.helpers.ip(address)
> 
> ---
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        self.device.send_command(cmd)
> >   +
> > -    def load_merge_candidate(self, filename=None, config=None):
> >   +
> > -        dst = "%s/tmp/%s" % (os.environ['HOME'], filename)
> > -        copyfile(filename, dst)
> > -        cmd = "oscmd rm /var/config/vcs/scripts/_candidate.cfg"
> > -        self.device.send_command(cmd)
> > -        cmd = "copy scp://%s:%s@%s/tmp/%s flash://_candidate.cfg" \
> > -            % (EXPORT_USER, EXPORT_PASSWORD, EXPORT_HOST, filename)
> >   +
> > -        self.device.send_command(cmd)
> >   +
> > -    def rollback_config(self):
> >   +
> > -        print "Reloading previous checkpoint ..."
> 
> ## Remove
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        lines = self.device.send_command(cmd)
> > -        lines = lines.split('\n')
> >   +
> > -        mac_address_table = []
> > -        # Skip the first four lines which is the header
> > -        lines = lines[1:-1]
> >   +
> > -        for line in lines:
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 7:
> > -                vlanid, tt, mac_address, type, state, port_type, port = \
> > -                    line.split()
> > -                entry = {
> > -                    'vlanid': vlanid,
> > -                    'mac+address': mac_address,
> 
> ## mac_address instead of mac+address
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        cmd = "show mac-address-table"
> > -        lines = self.device.send_command(cmd)
> > -        lines = lines.split('\n')
> >   +
> > -        mac_address_table = []
> > -        # Skip the first four lines which is the header
> > -        lines = lines[1:-1]
> >   +
> > -        for line in lines:
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 7:
> > -                vlanid, tt, mac_address, type, state, port_type, port = \
> > -                    line.split()
> > -                entry = {
> > -                    'vlanid': vlanid,
> 
> ## Just vlan is the name of the field
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > -        mac_address_table = []
> > -        # Skip the first four lines which is the header
> > -        lines = lines[1:-1]
> >   +
> > -        for line in lines:
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 7:
> > -                vlanid, tt, mac_address, type, state, port_type, port = \
> > -                    line.split()
> > -                entry = {
> > -                    'vlanid': vlanid,
> > -                    'mac+address': mac_address,
> > -                    'type': type,
> > -                    'state': state,
> > -                    'port_type': port_type,
> 
> Field port_type should not exist: https://github.com/napalm-
> 
> ## automation/napalm-base/blob/develop/napalm_base/test/models.py#L194
> 
> In napalm_brocade/brocade.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > +
> > -        mac_address_table = []
> > -        # Skip the first four lines which is the header
> > -        lines = lines[1:-1]
> >   +
> > -        for line in lines:
> > -            if len(line) == 0:
> > -                return {}
> > -            if len(line.split()) == 7:
> > -                vlanid, tt, mac_address, type, state, port_type, port = \
> > -                    line.split()
> > -                entry = {
> > -                    'vlanid': vlanid,
> > -                    'mac+address': mac_address,
> > -                    'type': type,
> > -                    'state': state,
> 
> state does not exist either: https://github.com/napalm-
> 
> ## automation/napalm-base/blob/develop/napalm_base/test/models.py#L194
> 
> In napalm_brocade/brocade.py~
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > @@ -0,0 +1,428 @@
> > +#
> 
> ## Remove this file, please
> 
> In napalm_brocade/utils/interactive/interactive.py
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387
> :
> 
> > @@ -0,0 +1,183 @@
> > +# Sample script to demonstrate loading a config for a device.
> 
> Remove this file please, this should be covered in the tests.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/napalm-automation/napalm-brocade/pull/1#pullrequestreview-3588387,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ADCK158WjUYB1_lwY0J0ay_cBXUoHVokks5qysqagaJpZM4KTEs8
> .

## 

-Shiv

mirceaulinic commented:
Yes @shivharis we received your reply. There's no rush on this - take your time and let's do it properly when works the best for you.

mirceaulinic commented:
Hey @shivharis - any progress on this?

Cheers!

mirceaulinic commented:
Hey @shivharis... hope you are fine!
Are you still interested in contributing this new driver?

Thanks!

dbarrosop commented:
@shivharis are you still working on this?

shivharis commented:
Hi Mircea,

Yes I am still interested in contributing to this driver. I got diverted
into writing/generating the switch API that I was planning on using for
this effort. The switch API is derived from the YANG model of the switch
and we have written some tools to auto-generate the API for NOS and SLX at
this this time.

I plan to switch to napalm soon after this effort - and use what we are
currently doing in this effort.

Thanks for your patience.

Shiv


On Tue, Jan 31, 2017 at 6:09 PM, Mircea Ulinic <notifications@github.com>
wrote:

> Hey @shivharis <https://github.com/shivharis>... hope you are fine!
> Are you still interested in contributing this new driver?
>
> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/napalm-automation/napalm-brocade/pull/1#issuecomment-276553836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADCK1-wtoBaTNF09RhvAuLyDwKBGr0vxks5rX-lkgaJpZM4KTEs8>
> .
>



-- 
-Shiv

dbarrosop commented:
Ok, great, keep us posted : )

ancker010 commented:
Is this still in progress?

dbarrosop commented:
I will assume there is no progress on this one. If nobody wants to take care of this I guess we will phase out this repo.

mirceaulinic commented:
Looks like this project has been abandoned, if there are no further comments in the next couple of days, I'll remove the repository.
